src/LazyOperations/CartesianProduct.jl

```diff
@@ -294,3 +294,34 @@ end
 function concretize(cp::CartesianProduct)
     return cartesian_product(concretize(cp.X), concretize(cp.Y))
 end
+
+"""
+    project(cp::CartesianProduct{N, Interval{N, IA.Interval{N}}, <:AbstractHyperrectangle{N}}, vars::AbstractVector{Int}) where {N}
```

Comment:
`vars` is called `block` in the documentation. I suggest to use the same name that is used in other methods.